Frontend/DinoTracker/tests/unit/controllers/login-test.ts

```diff
@@ -0,0 +1,39 @@
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import { click, currentURL, visit } from '@ember/test-helpers';
+import Service from '@ember/service';
+
+class MockApolloRepositoryService extends Service
+{
+  login(variables: { username: string, password: string })
+  {
+    return variables.username == "user1" && variables.password == "1234";
+  }
+}
+
+module('Unit | Controller | login', function(hooks) {
+  setupTest(hooks);
+
+  test('home button redirects to home page', async function(assert) {
+    await visit('/login');
+
+    await click('.button a.link');
+
+    assert.equal(currentURL(), '/');
+  });
+
+  test('successful login redirects to staff page', async function(assert) {
```

Comment:
This is fine but FYI for MICIP or the future, I've tried to keep our tests UI focused (testing from the perspective of the DOM) on the frontend. I'm not a big fan of testing methods on the controllers or components cause those can easily be refactored and that means you have to re-write your tests... The UI won't be refactored as much though and by testing your UI, you will also end up testing those methods because the UI depend on them. It kills 2 birds with 1 stone and you don't have to refactor tests as much.